File: tokumaru_sample/ch04-2/Match1.java
//javac Match1.java
//java Match1

import static java.lang.System.out;

class  Match1{
    public static void main(String args[]){

        System.out.println("制御文字以外を表す正規表現 \n");

        String param = "11";
        out.println("正常値\n");
        out.println(String.format("入力文字列:　%s \n", param));

        checkParamNormal(param);
        checkParamPCc(param);
        out.println("==============\n");
        out.println("5文字以上\n");
        param = "aaaaaaa";
        out.println(String.format("入力文字列:　%s \n", param));

        checkParamNormal(param);
        checkParamPCc(param);
        out.println("==============\n");
        out.println("制御文字列\n");
        param = "aa\t";
        out.println(String.format("入力文字列:　%s \n", param));

        checkParamNormal(param);
        checkParamPCc(param);
        out.println("==============\n");
    }



    //引数に渡した正規表現とこの文字列が一致する場合はtrue、それ以外の場合はfalse
    private static void checkParamNormal(String param){
        //1~5文字のアルファベットと数字
        if (param.matches("\\A[a-z0-9]{1,5}\\Z") == true) {
            out.println("Normal検証OK！\n");
        } else {
            out.println("Normal検証NG！\n");
        }
    }

    private static void checkParamPCc(String param) {
        if (param.matches("\\A\\P{Cc}{1,5}\\z") == true) {
            out.println("PCc検証OK！\n");
        } else {
            out.println("PCc検証NG！\n");
        }

    }

}

File: tokumaru_sample/ch04-2/match1.rb


p "制御文字以外を表す正規表現 \n";

#!~演算子は、マッチしなかったときにtrueを、マッチしたときにfalseを返す

def checkParamNormal(param)
    #1~5文字のアルファベットと数字
    if param !~ /\A[a-z0-9]{1,5}\Z/ then
        print("Normal検証NG！\n");
    else
        puts "Normal検証OK！\n";
    end
end

def checkParamPCc(param)
    if param !~ /\A\P{Cc}{1,5}\z/ then
        print("PCc検証NG！\n");
    else
        puts "PCc検証OK！\n";
    end
end

s = 11;
print "正常値\n";
print "入力文字列:　#{s} \n";

puts s.class
#マッチしたかどうか返す
p s !~ /\A[a-z0-9]{1,5}\z/
#これはマッチした位置を返す
p s =~ /\A[a-z0-9]{1,5}\z/
s = s.to_s
puts s.class

checkParamNormal(s);
checkParamPCc(s);
print "==============\n";
print "5文字以上\n";
s = "aaaaaaa";
print "入力文字列:　#{s} \n";
checkParamNormal($s);
checkParamPCc($s);
print "==============\n";
print "制御文字列\n";
s = "aa\t";
print "入力文字列:　#{s} \n";
checkParamNormal($s);
checkParamPCc($s);
print "==============\n";


